sonar-scanner-engine/src/main/java/org/sonar/scanner/report/MetadataPublisher.java

```diff
@@ -47,15 +57,25 @@
   private final ScannerPluginRepository pluginRepository;
   private final BranchConfiguration branchConfiguration;
 
+  @Nullable
+  private final ScmConfiguration scmConfiguration;
+
   public MetadataPublisher(ProjectAnalysisInfo projectAnalysisInfo, InputModuleHierarchy moduleHierarchy, Configuration settings,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)